### PR TITLE
Cleaned up the setup logic

### PR DESCRIPTION
**Pull request by MatthiasWiesmann**
Moved all the initialisation logic into a __init__.py file. 
Migrated all the code in software/util to use this mechanism.

### CODE REVIEW

File: software/util/buildsite.py [hunk 1]
```diff
@@ -1,44 +1,32 @@
...
-for dir in ['software/util','docs','software/gcloud','data']:
-    if not os.path.isdir(dir):
-        print('\nRequired directory "%s" not found - Exiting\n' % dir)
-        sys.exit(os.EX_CONFIG)
-
 # Import standard python libraries
 import argparse
 import glob
+import os
+import os
```
Comment (jlndr) [inline software/util/buildsite.py:8]: nit: double import

File: software/util/buildsite.py [hunk 2]
```diff
@@ -1,44 +1,32 @@
...
 import re
 import shutil
 import subprocess
-import textutils
+import sys
+import sys
```
Comment (jlndr) [inline software/util/buildsite.py:13]: nit: double import

File: software/__init__.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        absolute_path = os.path.join(os.getcwd(), path)
+        if not os.path.isdir(absolute_path):
+            sys.stderr.write('Required directory "%s" not found - Exiting\n' % absolute_path)
+            sys.exit(os.EX_CONFIG)
+        sys.path.insert(1, absolute_path)
+
+    _INITIALIZED = True
+
+
+def CheckWorkingDirectory():
```
Comment (jlndr) [inline software/__init__.py:32]: Did you intend to call this after `Setup()`? I can't see it referenced

File: software/util/buildsite.py (hunk 1 above)
Comment (MatthiasWiesmann) [reply software/util/buildsite.py:8]: Fixed.

File: software/__init__.py (hunk 3 above)
Comment (MatthiasWiesmann) [reply software/__init__.py:32]: I meant to call it from the buildsite.py file. Fixed.

**Review by jlndr — APPROVED** [review]
LGTM

**Review by alex-jansen — APPROVED** [review]

**Pull request merged**